web/src/main/java/org/springframework/security/web/firewall/ClientErrorRequestRejectedHandler.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.web.firewall;
+
+import java.io.IOException;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * A simple implementation of {@link RequestRejectedHandler} that sends and error with status code 400.
+ *
+ * @author Leonard Brünings
+ * @since 5.2
+ */
+public class ClientErrorRequestRejectedHandler implements RequestRejectedHandler {
+	private static final Log logger = LogFactory.getLog(ClientErrorRequestRejectedHandler.class);
+
+	@Override
+	public void handle(HttpServletRequest request, HttpServletResponse response,
+			RequestRejectedException requestRejectedException) throws IOException {
+		if (logger.isDebugEnabled()) {
+			logger.debug("Rejecting request due to: " + requestRejectedException.getMessage(),
+					requestRejectedException);
+		}
+		response.sendError(HttpServletResponse.SC_BAD_REQUEST);
```

Comment:
I think we should allow the http status to be injected through the constructor rather than hard coding the status. We should also rename the class to reflect this change. Finally, a default constructor with bad request as the default also makes a lot of sense